pkg/cloud/services/secretsmanager/secret.go

```diff
@@ -53,26 +53,44 @@ func (s *Service) Create(m *scope.MachineScope, data []byte) (string, int32, err
 	})
 
 	// Build the prefix.
-	prefix := path.Join(entryPrefix, string(uuid.NewUUID()))
-
+	prefix := m.GetSecretPrefix()
+	if prefix == "" {
+		prefix = path.Join(entryPrefix, string(uuid.NewUUID()))
+	}
 	// Split the data into chunks and create the secrets on demand.
-	var err error
+	var errs []error
 	chunks := int32(0)
 	splitBytes(data, maxSecretSizeBytes, func(chunk []byte) {
 		name := fmt.Sprintf("%s-%d", prefix, chunks)
-		_, callErr := s.scope.SecretsManager.CreateSecret(&secretsmanager.CreateSecretInput{
-			Name:         aws.String(name),
-			SecretBinary: chunk,
-			Tags:         converters.MapToSecretsManagerTags(tags),
-		})
-		if callErr != nil {
-			err = kerrors.NewAggregate([]error{callErr})
-			return
+		var done bool
+		for !done {
+			_, createErr := s.scope.SecretsManager.CreateSecret(&secretsmanager.CreateSecretInput{
```

Comment:
On a semi-related note, do we need to worry about eventual consistency on the instance side with retrieving the certs we are creating?

Reply:
hmm, think i've found something. thanks for that intervention.

/hold